library/extractor/src/main/java/com/google/android/exoplayer2/text/dvb/DvbTeletextParser.java

```diff
@@ -0,0 +1,513 @@
+package com.google.android.exoplayer2.text.dvb;
+
+import android.os.Build;
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+import android.text.style.BackgroundColorSpan;
+import android.text.style.ForegroundColorSpan;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.text.Cue;
+import com.google.android.exoplayer2.util.Log;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** Parses {@link Cue}s from a DVB Teletext subtitle bitstream. */
+/* package */ final class DvbTeletextParser {
```

Comment:
Does this need to be its own class? It kinda feels like it could be in-lined into `DvbTeletextDecoder`

Reply:
Well, I applied exactly the same logic as the existing `DvbParser`/`DvbDecoder` classes.